server/src/main/java/org/elasticsearch/action/search/SearchQueryThenFetchAsyncAction.java

```diff
@@ -77,10 +77,7 @@ class SearchQueryThenFetchAsyncAction extends AbstractSearchAsyncAction<SearchPh
         // register the release of the query consumer to free up the circuit breaker memory
         // at the end of the search
         addReleasable(resultConsumer);
-
-        if (progressListener != SearchProgressListener.NOOP) {
```

Comment:
Why'd you remove this check, this call isn't free for the no-op listener (due to collecting the list of shards that O complained about above :D), why remove the optimisation?